libsplinter/src/circuit/template/yaml_parser/v0_1.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct YamlCircuitCreateTemplate {
+    version: String,
+    args: Vec<YamlRuleArgument>,
+    rules: YamlRules,
+}
+
+impl YamlCircuitCreateTemplate {
+    pub fn version(&self) -> String {
```

Comment:
Convention in rust is to return references from getter methods when possible, that way the calling code can decide if a clone is necessary.

Reply:
Addressed